src/main/java/edu/ucsb/nceas/mdqengine/dispatch/Dispatcher.java

```diff
@@ -79,85 +102,136 @@ public Result dispatch(Map<String, Object> variables, String code) throws Script
+				log.debug("result is class: " + var_r.getClass());
+
+				dr = (Result) var_r;
 			} else {
-				
-				// if no explicit output included in structure
-				if (res !=null) {
-					dr.setOutput(new Output(res.toString()));
+				// try to find other result items from python checks
+				try {
+					var_py = engine.get("call()"); // run the python function
+				} catch (Exception e) {
+					log.error(e.getMessage());
+					log.error(e.getStackTrace());
+					dr.setOutput(new Output("ERROR: " + e.getMessage())); // catch the python stack trace
+					dr.setStatus(Status.valueOf("ERROR"));
+				}
+				// try to get the global output variable from python
+				try {
+					var_py = engine.get("output");
+				} catch (Exception e) {
+					// catch this silently since we are just fishing
+					// the no result case is handled later
+				}
+				// save the output
+				if (var_py != null && !var_py.toString().equals("<unbound>")) {
+					dr.setOutput(new Output(var_py.toString()));
+					var_py = null;
 				}
-				
-				// try to find other result items
-				var = engine.get("output");
-				if (var != null && !var.toString().equals("<unbound>")) {
-					dr.setOutput(new Output(var.toString()));
+				// try to get the global status variable from python
+				try {
+					var_py = engine.get("status");
+				} catch (Exception e) {
+					// catch this silently since we are just fishing
+					// the no result case is handled later
 				}
-				var = engine.get("status");
-				if (var != null && !var.toString().equals("<unbound>")) {
-					dr.setStatus(Status.valueOf(var.toString()));
+				// save the status
+				if (var_py != null && !var_py.toString().equals("<unbound>")) {
+					dr.setStatus(Status.valueOf(var_py.toString()));
 				} else {
-					// assume a true result means that the test was successful
-					if (res !=null) {
-						boolean interpolatedStatus = Boolean.valueOf(res.toString());
-						if (interpolatedStatus) {
-							dr.setStatus(Status.SUCCESS);
-						} else {
-							dr.setStatus(Status.FAILURE);
-						}
-					}
+					// if we haven't found anything at this point it probably failed
+					dr.setStatus(Status.FAILURE);
 				}
 			}
 		}
-		
+
 		// harvest all other vars for downstream dispatchers
 		bindings = engine.getBindings(ScriptContext.ENGINE_SCOPE);
-		
+
 		return dr;
-		
+
 	}
-	
+
+	/**
+	 * Checks if the environment is supported by the ScriptEngine.
+	 *
+	 * @param env The environment string to check for compatibility.
+	 * @return {@code true} if the specified environment is supportedScriptEngine
+	 *         {@code false} otherwise.
+	 */
 	public boolean isEnvSupported(String env) {
 		String currentEnv = engine.getFactory().getLanguageName();
 		log.debug("currentEnv=" + currentEnv);
 		return env.equalsIgnoreCase(currentEnv);
 	}
-	
-	protected Dispatcher() {}
 
+	public void close() {
+		if (engine instanceof JepScriptEngine) {
+			JepScriptEngine jepEngine = (JepScriptEngine) engine;
```

Comment:
Today I learned from my IDE..

lines 187 & 188 can be replaced simply by:

```java
		if (engine instanceof JepScriptEngine jepEngine) {
```

How cool is that?? 😁

Reply:
ooh nice!